### PR TITLE
feat: pv & pvc

### LINKED ISSUE #176: Construct for PersistentVolumeContainer (PVC)
There does not seem to be a way to attach a PVC to a container in a  deployment with cdk8s+.  My goal is to persist data added to a DB. I know I could go with a stateful set, but that is beyond what my simple sample use case requires. Basically, I want to generate something like this (partial manifest shown below):

```yaml
       ..... 
         volumeMounts:
        - mountPath: /var/lib/mysql
          name: mariadb-claim0
      volumes:
      - name: mariadb-claim0
        persistentVolumeClaim:
          claimName: mariadb-claim0
     .....
```

In theory, I can define the volume_mount that contains a Volume and a PVC:

```python
import cdk8s_plus_22 as kplus
from imports import k8s

mariadb_pvc = k8s.KubePersistentVolumeClaim(mariaDB_chart,
                                   'mariadbPvc',
                                   spec=k8s.PersistentVolumeClaimSpec(access_modes=['ReadWriteOnce'],
                                                        resources=k8s.ResourceRequirements(
                                                         requests={
                                                                  "storage": k8s.Quantity.from_string(
                                                                  "100Mi")})))

vm = kplus.VolumeMount(path="/var/lib/mysql", volume=k8s.Volume(
                        name="test",
                        persistent_volume_claim=k8s.PersistentVolumeClaimVolumeSource(claim_name="mariadbPvc")
                       ))
```

I then try to add the volume mount to the container:

```python
kplus.ContainerProps(image='mariadb',
           name="mariadbcont",
           port=3306,
           volume_mounts=[vm]
          )
```

But  I get the following error:

```
raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Expected object reference, got {"$jsii.struct":{"fqn":"k8s.Volume","data":{"name":"test","awsElasticBlockStore":null,"azureDisk":null,"azureFile":null,"cephfs":null,"cinder":null,"configMap":null,"csi":null,"downwardApi":null,"emptyDir":null,"ephemeral":null,"fc":null,"flexVolume":null,"flocker":null,"gcePersistentDisk":null,"gitRepo":null,"glusterfs":null,"hostPath":null,"iscsi":null,"nfs":null,"persistentVolumeClaim":{"$jsii.struct":{"fqn":"k8s.PersistentVolumeClaimVolumeSource","data":{"claimName":"mariadbPvc","readOnly":null}}},"photonPersistentDisk":null,"portworxVolume":null,"projected":null,"quobyte":null,"rbd":null,"scaleIo":null,"secret":null,"storageos":null,"vsphereVolume":null}}}
```

I think the ContainerProps only take volume_mounts from config files or emptyDir at this point.

### PR DESCRIPTION
**Pull request by iliapolo**
Fixes https://github.com/cdk8s-team/cdk8s-plus/issues/176

Corresponding docs PR should explain the API more clearly: https://github.com/cdk8s-team/cdk8s/pull/889

### CODE REVIEW

File: src/pv.ts [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+   * Contains all ways the volume can be mounted.
+   *
+   * @see https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes
+   * @default - No access modes.
+   */
+  readonly accessModes?: PersistentVolumeAccessMode[];
+
+  /**
+    * What is the storage capacity of this volume.
+    *
```
Comment (Chriscbr) [inline src/pv.ts]: nit: docstring formatting

File: src/pv.ts [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+
+  /**
+    * Storage class this volume belongs to.
+    */
+  public readonly storageClassName?: string;
+
+  /**
+    * Access modes requirement of this claim.
+    */
+  public readonly accessModes?: PersistentVolumeAccessMode[];
```
Comment (Chriscbr) [inline src/pv.ts]: Should we be using a getter here instead (so that we only expose an immutable copy)?

**Review by Chriscbr — APPROVED** [review]
🤩  Awesome work! I left some comments about the API on the other PR but otherwise I think the API looks really solid.

File: src/pv.ts (hunk 2 above)
Comment (iliapolo) [reply src/pv.ts]: Yeap, good catch 👍

File: src/pv.ts (hunk 1 above)
Comment (iliapolo) [reply src/pv.ts]: 👍

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/571)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/572)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**